src/http/parser.rs

```diff
@@ -0,0 +1,290 @@
+use std::collections::{HashMap, LinkedList};
+
+use serde::de::{DeserializeOwned, Error};
+use serde_json::{Map, Value};
+
+use crate::async_graphql_hyper::GraphQLRequestLike;
+use crate::blueprint::Definition;
+use crate::parser::de::{de_kebab, next_token, to_json, to_json_str};
+
+#[derive(Debug, serde::Deserialize, Default)]
+pub struct Parser {
+  operation: String,
+  arguments: Option<String>,
+  selections: Option<String>,
+}
+
+impl Parser {
+  pub fn from_path(path: &str) -> anyhow::Result<Parser> {
+    let path = urlencoding::decode(&de_kebab(path)).unwrap().replace('\\', "");
+    let mut root = String::new();
+    let split = path.split('?').collect::<Vec<&str>>();
+    let path = split.last().unwrap();
+    match serde_qs::from_str::<Map<String, Value>>(path) {
+      Ok(p) => {
+        let mut parser = Self::default();
+        let rootname = split.first().unwrap();
+        let mut to_camel = false;
+        for char in rootname.chars().skip(4) {
+          match char {
+            '/' => (),
+            '-' => {
+              to_camel = true;
+            }
```

Comment:
is that de_kebab logic again?

Reply:
ah yes I forgot that I already switched to a function, I'll remove it